src/libraries/Native/Windows/CMakeLists.txt

```diff
@@ -58,72 +62,54 @@ endif ()
 
 # enable control-flow-guard support for native components
 add_compile_options(/guard:cf) 
-set(__LinkArgs "${__LinkArgs} /GUARD:CF")
+list(APPEND __SharedLinkArgs /GUARD:CF)
 
 # Statically linked CRT (libcmt[d].lib, libvcruntime[d].lib and libucrt[d].lib) by default. This is done to avoid
 # linking in VCRUNTIME140.DLL for a simplified xcopy experience by reducing the dependency on VC REDIST.
 #
 # For Release builds, we shall dynamically link into uCRT [ucrtbase.dll] (which is pushed down as a Windows Update on downlevel OS) but
 # won't do the same for debug/checked builds since ucrtbased.dll is not redistributable and Debug/Checked builds are not
 # production-time scenarios.
-if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL "RELEASE")
-    add_compile_options(/MT)
-    add_compile_options(/GL)
-    add_compile_options(/O1)
-    set(__LinkLibraries "${__LinkLibraries} libcmt.lib")
-    set(__LinkLibraries "${__LinkLibraries} libvcruntime.lib")
-    set(STATIC_UCRT_LIB "libucrt.lib")
-    set(DYNAMIC_UCRT_LIB "ucrt.lib")
-else()
-    add_compile_options(/MTd)
-    set(__LinkLibraries "${__LinkLibraries} libcmtd.lib")
-    set(__LinkLibraries "${__LinkLibraries} libvcruntimed.lib")
-    set(STATIC_UCRT_LIB "libucrtd.lib")
-    set(DYNAMIC_UCRT_LIB "ucrtd.lib")
-endif()
+add_compile_options(/MT$<$<OR:$<CONFIG:DEBUG>,$<CONFIG:CHECKED>>:d>)
```

Comment:
Nit: i'd probably select whole names like `libcmt.lib` vs. `libcmtd.lib`.  
Selecting just the presence of `d` makes generators shorter, but with full names it is a bit easier to figure what is going on.